src/CoreBundle/private/library/classes/TCMSLogChange.class.php

```diff
@@ -1779,6 +1779,51 @@ public static function escapeJSOutput($str)
         return $str;
     }
 
+    /**
+     * This method lets old updates insert shop system pages. The table "shop_system_page" has been removed in shop-165.inc.php and
+     * old updates would fail when triggered afterwards.
+     *
+     * You should not use this method in new packages, which already know about the change.
+     *
+     * @param $shop_id
+     * @param $name_internal
+     * @param $name
+     * @param string $cms_tree_id
+     *
+     * @deprecated since 6.2.0 - no longer used.
+     */
+    public static function addShopSystemPage($shop_id, $name_internal, $name, $cms_tree_id = '', $id = null)
+    {
+        if (self::TableExists('shop_system_page')) {
+            $query = "INSERT INTO `shop_system_page`
+                      SET `shop_id` = '".$shop_id."',
+                          `name_internal` = '".$name_internal."',
+                          `name` = '".$name."',
+                          `cms_tree_id` = '".$cms_tree_id."'";
+            if (null !== $id) {
+                $query .= ", `id`='".MySqlLegacySupport::getInstance()->real_escape_string($id)."'";
+            }
+            self::_RunQuery($query, __LINE__);
+        } else {
+            $shop = TdbShop::GetNewInstance();
+            if ($shop->Load($shop_id)) {
+                $portals = $shop->GetFieldCmsPortalIdList();
+                foreach ($portals as $portalid) {
+                    $query = "INSERT INTO `cms_portal_system_page` SET
+                                `cms_portal_id` =   '".MySqlLegacySupport::getInstance()->real_escape_string($portalid)."',
```

Comment:
Is this copied from somewhere?

Otherwise on could use "database_connection" for the escaping.

Reply:
See above.